### ISSUE TITLE
Bazel Build from source relies on *which*

### ISSUE THREAD

**Issue by cinqs** [bazelbuild/bazel#3535]
### Description of the problem / feature request / question:

If OS lacking the package *which*, there will be an error:

```console
    ERROR: cannot locate GNU bintools; check your PATH.
       (You may need to run 'export PATH=/bin:/usr/bin:$PATH)'
```

### If possible, provide a minimal example to reproduce the problem:

```sh
yum remove which
bash /opt/bazel/compile.sh
```

### Environment info
* Operating System: Docker CentOS 7

* Bazel version (output of `bazel info release`): 0.5.3

* If `bazel info release` returns "development version" or "(@non-git)", please tell us what source tree you compiled Bazel from; git commit hash is appreciated (`git rev-parse HEAD`):

I am still compiling the source...

###  Have you found anything relevant by searching the web?

installed the *which* solved the problem

```sh
yum install -y which
```

### Anything else, information or logs or outputs that would be helpful?
(If they are large, please upload as attachment or provide link).

**Comment by mhlopko** [comment]
Hello @cinqs,
what is the problem you have here? That bazel relies on which, or that all the requirements are not well documented?

**Comment by cinqs** [comment]
Yes, by *which*, I mean the package *which*, on CentOS, you use 

    yum install which

to get it installed.

**Comment by mhlopko** [comment]
Ok thanks, assigning to Damien to decide if/where we should document basic requirements to build bazel, and if we should make it per distribution on linux, and if Centos is one of the explicitly supported distributions.

**Comment by cinqs** [comment]
I don't want to start an argument on which is the best Linux distribution, CentOS or Ubuntu... 

But why don't you officially support CentOS? Any particular reason you choose Ubuntu as the first choice? 

As from what I saw, Google developers' product commonly chose Ubuntu as the default OS to support. Why?

**Comment by mhlopko** [comment]
I apologize if my response came out that way. I didn't want to express any preference or any subjective criteria on choosing the supported distributions. Of course we want to support the distributions our users are using, but this comes with a cost and needs to be evaluated. Thus assigning to Damien.

**Comment by damienmg** [comment]
@cinqs: we do support CentOS as a bootstrapping environment and [test for it](https://github.com/bazelbuild/bazel/blob/master/src/test/docker/Dockerfile.centos6.7). We do need `which` to bootstrap indeed but not to run bazel. We do not provide RPM package yet. 

We choosed Ubuntu because that was the default distribution inside Google.

I consider this a documentation issue, we should clearly add the requirements to build on centos. Also probably add a docker test for centos 7.

/cc @michelleirvine

**Comment by michelleirvine** [comment]
I've created #3593 pls comment there about doc for installing on CentOS.

**Comment by aehlig** [comment]
Is that issue still present after 27db2d8f0923751f99ea824096188153d49839ea?

**Comment by cinqs** [comment]
@aehlig , Hi, I didn't have the chance yet to test on commit 27db2d8, but from the scripts change in `compile.sh`, it seems fine. The issues should be gone with this commit.
I filed this issue because either your OS is lacking `bintools` or `which` could cause the result of showing:

    ERROR: cannot locate GNU bintools; check your PATH.

And this should be avoided.

And more, the latest version released is `0.5.4`, which didn't contain the commit 27db2d8. means still having the problem

**Comment by cinqs** [comment]
And this can be closed I think
